src/mixins/labels.js

```diff
@@ -95,6 +96,71 @@ export const LabelsMixin = (superclass) =>
             });
         }
 
+        /**
+         * This function makes a hash map from atom names.
+         * Used it for sorting all atoms by names and then quckly get it for creating a label points
+         * @returns {Object.<string, Array.<Three.InstancedMesh>>} atomsHashMap
+         */
+        makeHashMapFromAtomNames() {
+            const atomsHashMap = {};
+            this.structureGroup.children.forEach((group) => {
```

Comment:
Why not just get the atomsGroup directly to make it more efficient?

Reply:
Because we can have many atom groups

Reply:
Investigated this one, apply small optimization here, not sure if I am really can make it more efficient then now.